joda-money/src/main/java/tools/jackson/datatype/jodamoney/MoneyDeserializer.java

```diff
@@ -35,6 +38,74 @@ public MoneyDeserializer() {
         this.amountConverter = requireNonNull(amountConverter, "amount converter cannot be null");
     }
 
+    @Override
+    public ValueDeserializer<?> createContextual(DeserializationContext ctxt, BeanProperty property) {
+        if (property == null) {
+            return this;
+        }
+
+        AmountRepresentation effectiveRepresentation = _resolveRepresentation(property, ctxt);
+
+        if (effectiveRepresentation == null || effectiveRepresentation == AmountRepresentation.DEFAULT) {
+            // Keep current converter (module-level default)
+            return this;
+        }
+
+        AmountConverter newConverter = _getConverterForRepresentation(effectiveRepresentation);
+        if (newConverter == this.amountConverter) {
+            return this;
+        }
+
+        return new MoneyDeserializer(newConverter);
+    }
+
+    private AmountRepresentation _resolveRepresentation(BeanProperty property, DeserializationContext ctxt) {
+        // Priority 1: @JsonMoney annotation
+        JsonMoney jsonMoney = property.getAnnotation(JsonMoney.class);
```

Comment:
Needs to have `null` check for `property`, either here or in caller: can be `null` in case of "root values", values passed directly to read/write (not referenced via actual property).